1_fetch/src/fetch_sites.R

```diff
@@ -0,0 +1,52 @@
+#' Download all NWIS stage sites within the bounding box of a supplied spatial object
+#'
+#' @param ind_file character file name where the output should be saved
+#' @param sf_object spatial object with IDs indicating the state name
+#' @param dates object from viz_config.yaml that specifies dates as string
+#' @param stream_params pcodes to use from NWIS
+fetch_sites_from_sf <- function(ind_file, sf_object, dates, stream_params) {
+
+  state_names <- sf_object$ID[!grepl(",", sf_object$ID)]
+  state_cds <- unique(dataRetrieval::stateCdLookup(state_names))
+
+  # Cast wide net for all NWIS sites with stage data that fall within that bbox
+  sites_df <- data.frame()
+  for(cd in state_cds) {
+    sites_df_cd <- dataRetrieval::whatNWISdata(stateCd = cd, parameterCd = stream_params$stage, service = "uv")
+    sites_df <- rbind(sites_df, sites_df_cd)
+  }
+  sites_df <- dplyr::filter(sites_df, site_tp_cd == "ST")
+  sites_df <- dplyr::filter(sites_df, end_date >= as.Date(dates$dates$end))
```

Comment:
I had to do this too, but any idea why we have to do `dates$dates$...` when we've already subset to `dates` in the `viz_config`?

Reply:
I think @aappling-usgs fixed that here, but I haven't had a chance to merge those changes in just yet! https://github.com/USGS-VIZLAB/vizstorm-GIF/pull/24/commits/a58c5513b8f2039b5d5159d93fba36a36cbbca08